deepdrivemd/data/stream/aggregator_reader.py

```diff
@@ -0,0 +1,261 @@
+    """
+
+    def next(self, ARW):
+        var = getattr(ARW, "d_" + self.name)
+        self.total.append(var[0])
+
+
+class AdiosReader:
+    """This class is used to read the next `N` steps from an adios stream.
+
+    Attributes
+    ----------
+    adios : adios2.adios2.ADIOS
+    io : adios2.adios2.IO
+    stream : adios2.adios2.Engine
+    """
+
+    def __init__(
+        self, fn: str, config: Path, stream_name: str, variables: List[StreamVariable]
+    ):
+        """Constructor
+
+        Parameters
+        ----------
+        fn: str
+            file name of bp file or sst socket (without sst extension)
+        config : Path
+            path to `adios.xml` file
+        stream_name : str
+            name of a stream in `adios.xml` file
+        """
+        self.adios = adios2.ADIOS(str(config), True)
+        self.io = self.adios.DeclareIO(stream_name)
+        self.stream = self.io.Open(fn, adios2.Mode.Read)
+
+        self.variables = variables
+        self.connections = {0: (self.adios, self.io, self.stream)}
+
+    def __del__(self):
+        """Destructor: clean the adios resources"""
+        self.stream.Close()
+        self.io.RemoveAllVariables()
+        self.adios.RemoveAllIOs()
+
+    def next(self, N: int) -> List:
+        """Read the next `N` steps of all variables.
+
+        Parameters
+        ----------
+        N : int
+            read that many steps
+
+        Returns
+        -------
+        List[int, List[np.ndarray], ...]
+             the first element is the actual number of steps read,
+             subsequent elements are lists of numpy arrays for each variable, for each time step
+        """
+
+        vvv = {}
+        for v in self.variables:
+            vvv[v.name] = (v.dtype, v.structure)
+            v.total = []
+
+        ARW = AdiosStreamStepRW(self.connections, vvv)
+
+        for i in range(N):
+            status = ARW.read_step(0)
+            if not status:
+                break
+            for v in self.variables:
+                v.next(ARW)
+        output = [i]
+        for v in self.variables:
+            output.append(v.total.copy())
+
+        return output
+
+
+class Streams:
+    """The class keeps `lastN` steps from each aggregator
+
+    Attributes
+    ----------
+    readers : Dict[str, AdiosReader]
+          a dictionary of `AdiosReader` indexed by the corresponding adios file name
+    positions : Dict[str, np.ndarray]
+    md5 : Dict[str, str]
+    steps : Dict[str, np.ndarray]
+    rmsds : Dict[str, np.ndarray]
+    cm : Dict[str, np.ndarray]
+    velocities : Dict[str, np.ndarray]
+    lastN : int
+         keep that many last steps from each aggregator
+    batch : int
+         up to how many steps to read from each adios file at a time
+
+    """
+
+    def __init__(
+        self,
+        file_list: List[str],
+        variables: List[StreamVariable],
+        config: str = "../aggregate/adios.xml",
```

Comment:
This variable should be updated to type Path since it is passed to AdiosReader which is expecting a Path type (other portions of the code also pass it a Path type, like in the CVAE main function.

It May also be worthwhile to update `file_list` to List[Path] to be consistent, but it is not a big deal. Also the `file_list` variable could be updated to `files` to match the other variables naming convention.

On my classes I like to use a PathLike argument so that they are flexible on what they can receive. This looks like:

```
PathLike = Union[Path, str]
```

Reply:
Done except for List[Path]